lib/matplotlib/axes/_axes.py

```diff
@@ -43,6 +43,24 @@
 # All the other methods should go in the _AxesBase class.
 
 
+def _make_axes_method(func):
+    """
+    Patch the qualname for functions that are directly added to Axes.
+
+    Some Axes functionality is defined in functions in other submodules.
+    These are simply added as attributes to Axes. As a result, their
+    ``__qualname__`` is e.g. only "table" and not "Axes.table". This
+    function fixes that.
+
+    Note that the function itself is patched, so that
```

Comment:
Alternatively, since you've wrapped all of them with this function, you could return a new function from here and not modify the original.

Reply:
For the scope of this PR, I don't care too much how this is implemented, I just need a correct `__qualname__` and changing is good enough.

Yes, you could create a new function, but that's an additional indirection. While negligible from a performance perspective, it shows up e.g. in stacktraces. I'm not even sure that this way of adding functionality to classes reasonable style and what other edge-case consequences that has. Alternatively, one could make the original functions private. But that's all for a different PR.